2016-05/may-05.md

```diff
@@ -0,0 +1,74 @@
+## May 5 ([discuss](https://github.com/reactjs/core-notes/pull/13))
+
+### Attendees
+
+* [Ben](https://twitter.com/soprano) (React)
+* [Dan](https://twitter.com/dan_abramov) (React)
+* [Jim](http://github.com/jimfb) (React)
+* [Paul](https://twitter.com/zpao) (React)
+* [Sebastian](https://twitter.com/sebmarkbage) (React)
+* [Shayne](https://github.com/shayne) (React Native)
+* [Tom](https://twitter.com/tomocchino) (React)
+
+### ReactPerf Rewrite
+
+* [Dan](https://twitter.com/dan_abramov) got his rewrite of `ReactPerf` [merged into React master](https://github.com/facebook/react/pull/6647).
+* It is mostly functionally identical to the old one but is written in a more maintainable way.
+* At Facebook, we don’t use the console `ReactPerf` API much so we’ll need community help testing this.
+* There is a corresponding [React Native PR](https://github.com/facebook/react-native/pull/7283) that depends on these changes.
+* The change removes `ReactPerf.measure()` wrappers and instead emits events (e.g. `onBeginLifeCycleTimer`).
+* Related PRs: [#6549](https://github.com/facebook/react/pull/6549), [#6612](https://github.com/facebook/react/pull/6612), [#6633](https://github.com/facebook/react/pull/6633), [#6046](https://github.com/facebook/react/pull/6046).
+* It will most likely go into `15.1.0-alpha` early next week.
+
+### TestUtils and Enzyme
+
+* Airbnb’s [Enzyme](https://github.com/airbnb/enzyme) is well maintained and supported.
+* Our [`TestUtils`](https://facebook.github.io/react/docs/test-utils.html) are not in a great shape, and most people prefer Enzyme.
+* Should Enzyme become the official `TestUtils`?
```

Comment:
This is one of the reasons why there is no decision on this yet. It's a tricky balance. “Testing primitives” is also vague: we *do* provide them today (`TestUtils`) but we are not sure that this is the primitives we want to expose. As noted in text, we’d rather see test runner *as a custom React renderer*. In this version of the future, React does not need to expose any hooks at all because the custom renderer has full control over components. This is probably where we want to go but the obstacle is we don't have a public API for renderers, and the internal APIs will be changing because of the work on incremental reconciler. Therefore writing custom renderers will be turbulent in the coming months. This was the reason React Native renderer code was pulled into React codebase—to allow faster iteration on the internals.

So none of this is clear yet. While `TestUtils` exist the way they do today, it doesn’t quite make sense to tell people to use something else yet. However in the future, it might make sense for Enzyme to become more tightly coupled with implementation details of React, at least until the renderer API shakes out. If this happens, and it gives us a way to deprecate `TestUtils` in the future, we’ll likely need to point to *something*—and I would expect that it’s likely we won’t have other test renderers at that point because creating a renderer right now is somewhat brittle and requires quite a bit of effort. In that case, maybe, recommending Enzyme officially might be sensible.

But we don’t know much yet. These are just thoughts aloud. We are talking through different options and did not choose any particular strategy.